### ISSUE TITLE
New api change 

### ISSUE THREAD

**Issue by yvoyer** [yvoyer/php-state#10]
```php
// in context
/**
  * @bar ClassState implement of State
  */
private $state; build using reflection allowed

public function isActive()
{
    return $this->state->hasAttribute('active');
}

public function publish()
{
    $this->state = $this->state->publish($this); // or transitTo('published')
}


// CustomState
@Embeddable 
final class ClassState implement StateMetadata // serves as interface to autocomplete 
{
    private $meta = [];

    public function publish($context, StateMachine $machine)
    {
        return $machine->get('published'); // State
    }
}
```

* should support Doctrine hydration using embeddable
* should be fast
* should allow easy update when persisted
* should be easy to use and extend when used as stand-alone

**Comment by yvoyer** [comment]
Add versioning to allow for migration support. 

When v1 discovered, migrate to next , update meta for save.

**Comment by yvoyer** [comment]
Implemented in #12

**Comment by yvoyer** [comment]
Doctrine usage fixed in #13
